### ISSUE TITLE
golangci-lint claims that import-unrelated comments are not gci-ed

### ISSUE THREAD

**Issue by jonathanwin** [golangci/golangci-lint#4040]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've read the typecheck section of the FAQ (https://golangci-lint.run/usage/faq/#why-do-you-have-typecheck-errors).
- [X] Yes, I've tried with the standalone linter if available (e.g., gocritic, go vet, etc.). (https://golangci-lint.run/usage/linters/)

### Description of the problem

golangci-lint claims that import-unrelated comments are not gci-ed, but:
* standalone gci does not mind
* gci should not be linting comments at all (except in import blocks)

While comments could indeed be prettier, maybe we don't agree with the suggested formatting, and why is golangci-lint saying that gci is complaining?

```console
$ gci --version
gci version 0.11.0
# same results in version 0.10.0

$ gci diff --skip-generated -s standard -s default ./gci.go
# no output

$ golangci-lint --version
golangci-lint has version 1.54.1 built with go1.21.0 from a9378d9b on 2023-08-11T12:49:16Z
# same results in version 1.53.3 built with go1.20.5 from 2dcd82f3 on 2023-06-15T10:50:11Z

$ gci diff --skip-generated -s standard -s default ./gci.go
# no output
```

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version 1.54.1 built with go1.21.0 from a9378d9b on 2023-08-11T12:49:16Z
```

</details>

### Configuration

<details>

no .golangci-lint configuration file

```console
golangci-lint run --disable-all --enable gci ./gci.go 
```

</details>

### Go environment

<details>

```console
$ go version && go env
go version go1.18.3 linux/amd64
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/jonathan/.cache/go-build"
GOENV="/home/jonathan/.config/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/jonathan/go/pkg/mod"
GONOPROXY="test.com"
GONOSUMDB="test.com"
GOOS="linux"
GOPATH="/home/jonathan/go"
GOPRIVATE="test.com"
GOPROXY="direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.18.3"
GCCGO="gccgo"
GOAMD64="v1"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/jonathan/go/src/test-gci/go.mod"
GOWORK=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build2961642998=/tmp/go-build -gno-record-gcc-switches"

```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
INFO [config_reader] Config search paths: [./ /home/jonathan/go/src/test-gci /home/jonathan/go/src /home/jonathan/go /home/jonathan /home /] 
INFO [lintersdb] Active 6 linters: [errcheck gosimple govet ineffassign staticcheck unused] 
INFO [loader] Go packages loading at mode 575 (imports|name|deps|exports_file|types_sizes|compiled_files|files) took 227.484975ms 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 163.361µs 
INFO [linters_context/goanalysis] analyzers took 10.131689883s with top 10 stages: buildir: 8.051856741s, inspect: 682.640682ms, fact_deprecated: 297.802303ms, ctrlflow: 293.413844ms, printf: 236.262269ms, fact_purity: 174.190166ms, SA5012: 150.060166ms, nilness: 149.637296ms, typedness: 91.50975ms, SA9002: 278.595µs 
INFO [runner] processing took 3.381µs with stages: max_same_issues: 497ns, skip_dirs: 357ns, nolint: 310ns, max_from_linter: 164ns, uniq_by_line: 159ns, cgo: 155ns, fixer: 150ns, filename_unadjuster: 148ns, source_code: 145ns, path_prettifier: 145ns, identifier_marker: 141ns, max_per_file_from_linter: 140ns, exclude: 135ns, skip_files: 134ns, autogenerated_exclude: 131ns, exclude-rules: 129ns, sort_results: 73ns, severity-rules: 71ns, path_shortener: 70ns, diff: 66ns, path_prefixer: 61ns 
INFO [runner] linters took 3.402466293s with stages: goanalysis_metalinter: 3.402416842s 
INFO File cache stats: 0 entries of total size 0B 
INFO Memory: 38 samples, avg is 335.3MB, max is 547.6MB 
INFO Execution took 3.638024081s    



$ golangci-lint cache clean
$ golangci-lint run --disable-all --enable gci ./gci.go -v
INFO [config_reader] Config search paths: [./ /home/jonathan/go/src/test-gci /home/jonathan/go/src /home/jonathan/go /home/jonathan /home /] 
INFO [lintersdb] Active 1 linters: [gci]          
INFO [loader] Go packages loading at mode 575 (types_sizes|compiled_files|files|name|deps|exports_file|imports) took 283.506199ms 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 793.55µs 
INFO [linters_context/goanalysis] analyzers took 660.588µs with top 10 stages: gci: 653.632µs, typecheck: 6.956µs 
INFO [runner] Processors filtering stat (out/in): max_from_linter: 3/3, path_shortener: 3/3, path_prettifier: 3/3, skip_files: 3/3, skip_dirs: 3/3, autogenerated_exclude: 3/3, exclude: 3/3, exclude-rules: 3/3, severity-rules: 3/3, identifier_marker: 3/3, fixer: 3/3, path_prefixer: 3/3, sort_results: 3/3, filename_unadjuster: 3/3, max_same_issues: 3/3, source_code: 3/3, cgo: 3/3, nolint: 3/3, uniq_by_line: 3/3, diff: 3/3, max_per_file_from_linter: 3/3 
INFO [runner] processing took 690.248µs with stages: exclude-rules: 261.628µs, identifier_marker: 194.258µs, nolint: 89.889µs, autogenerated_exclude: 40.133µs, path_prettifier: 37.326µs, source_code: 36.214µs, skip_dirs: 12.892µs, max_same_issues: 3.619µs, uniq_by_line: 2.818µs, path_shortener: 2.208µs, max_from_linter: 2.03µs, cgo: 1.934µs, filename_unadjuster: 1.167µs, skip_files: 858ns, max_per_file_from_linter: 825ns, fixer: 537ns, severity-rules: 494ns, exclude: 433ns, diff: 382ns, sort_results: 371ns, path_prefixer: 232ns 
INFO [runner] linters took 148.69883ms with stages: goanalysis_metalinter: 147.68254ms 
gci.go:13: File is not `gci`-ed with --skip-generated -s standard -s default (gci)
//  * item 1
//  * item 2
//  * item 3
gci.go:24: File is not `gci`-ed with --skip-generated -s standard -s default (gci)
// 		nihil dolores eius ut illo consequatur itaque magnam. Dolore distinctio ut et. Aut consequuntur
gci.go:26: File is not `gci`-ed with --skip-generated -s standard -s default (gci)
//		fugiat nam. Possimus et dolorem quae.
INFO File cache stats: 1 entries of total size 645B 
INFO Memory: 6 samples, avg is 29.4MB, max is 44.1MB 
INFO Execution took 443.919305ms     
```

</details>

### A minimal reproducible example or link to a public repository

<details>
file: gci.go
```go
package main

import (
	"fmt"

	"github.com/cockroachdb/errors"
)

// Test comment
//
// test body
//
//  * item 1
//  * item 2
//  * item 3
func main() {
	foo()
	fmt.Printf("%v", errors.Errorf("foo"))
}

// lorem ipsum
// dolor sit amet
// Ratione sed provident natus corrupti dolore cupiditate minima quidem. Quis quia ut sed. Voluptatibus
// 		nihil dolores eius ut illo consequatur itaque magnam. Dolore distinctio ut et. Aut consequuntur
// Possimus non tempora et. Et in velit sed quibusdam non. Ut ducimus quod est. Non officia id maxime
//		fugiat nam. Possimus et dolorem quae.
// consequatur quae qui excepturi aliquam.
func foo() {
}

```

</details>

### Validation

- [X] Yes, I've included all information above (version, config, etc.).

**Comment by ldez** [comment]
Hello,

It's because your file is not well formatted (gofmt)

**Comment by jonathanwin** [comment]
Interesting...  `gofmt -d gci.go` produces no output, and `gofmt -w gci.go` does not change the file.
`go fmt ./...` is part of our regular checks

Maybe extra formatting rules have been added to golang between our current version in the problematic project (1.18.3) and the one used by golangci-lint (go1.21.0), but still:
* I think that golangci-lint should report this as a formatting issue, since it has nothing to do with the gci linter
* how come standalone gci doesn't complain?
* how come the other golangci-lint linters do not complain?

Thanks,
Jonathan

**Comment by ldez** [comment]
The way that gci can be integrated as a lib for us to format the code.
Only a few linters are implemented like that.
